wolfcrypt/src/asn.c

```diff
@@ -20970,48 +20975,79 @@ static int CopyValidity(byte* output, Cert* cert)
 #endif /* !WOLFSSL_ASN_TEMPLATE */
 #endif
 
+
+/* Simple name OID size. */
+#define NAME_OID_SZ     3
+
+/* Domain name OIDs. */
+static const byte nameOid[][NAME_OID_SZ] = {
+    { 0x55, 0x04, ASN_COUNTRY_NAME },
+    { 0x55, 0x04, ASN_STATE_NAME },
+    { 0x55, 0x04, ASN_STREET_ADDR },
+    { 0x55, 0x04, ASN_LOCALITY_NAME },
+    { 0x55, 0x04, ASN_SUR_NAME },
+    { 0x55, 0x04, ASN_ORG_NAME },
+    { 0x00, 0x00, ASN_DOMAIN_COMPONENT}, /* not actual OID - see dcOid */
+                                         /* list all DC values before OUs */
+    { 0x55, 0x04, ASN_ORGUNIT_NAME },
+    { 0x55, 0x04, ASN_COMMON_NAME },
+    { 0x55, 0x04, ASN_SERIAL_NUMBER },
+#ifdef WOLFSSL_CERT_EXT
+    { 0x55, 0x04, ASN_BUS_CAT },
```

Comment:
The index changes based on compile flags. Is this OK?
We can just always define this.

Reply:
Part of this work is a refactor of the code to not rely on on index.

Reply:
Only used in for loops. OK.